### ISSUE TITLE
Highlight: support same background settings as e.g. box

### ISSUE THREAD

**Issue by yochem** [typst/typst#2934]
### Description

Currently, [`highlight`](https://typst.app/docs/reference/text/highlight/) only support `extend` to modify the shape of the background, which extends in all directions. It would be nice if it had the same options as [`box`](https://typst.app/docs/reference/layout/box/):

- [stroke](https://typst.app/docs/reference/layout/box/#parameters-stroke)
- [radius](https://typst.app/docs/reference/layout/box/#parameters-radius)
- [inset](https://typst.app/docs/reference/layout/box/#parameters-inset)
- [outset](https://typst.app/docs/reference/layout/box/#parameters-outset)

### Use Case

Box is a rectangle and can't break over lines. Thus, highlighting long code like here on GitHub is not possible:

`Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim aeque doleamus animo`

This can be done with the highlight function, but it lacks the styling possibilities to make it really useful.

Another option would be to add this general styling of the background to the text function.

**Comment by deffi** [comment]
Note that on GitHub, the radius is not applied at the line break.

**Comment by deffi** [comment]
Related discussion with workaround: #2911

**Comment by deffi** [comment]
Also, the `extent` parameter for `highlight` is only applied in the horizontal direction. It might be useful if we could also have that in vertical direction, or different values for left and right.

**Comment by yochem** [comment]
> Also, the extent parameter for highlight is only applied in the horizontal direction. It might be useful if we could also have that in vertical direction, or different values for left and right.

Yes, just like inset and [outset](https://typst.app/docs/reference/layout/box/#parameters-outset), either a single value for all sides, or a dictionary

**Comment by sijow** [comment]
Maybe adding box properties to `highlight` is a bad idea... I understand `highlight` as a the Typst equivalent of a felt-tip pen: it never disturbs the layout and the bounds adapt to the text size.

Already `radius` and `stroke` have been added and it's not clear how they should behave when the text height is changing (see #5324).

I guess `highlight` is often used where people really want a breakable box, and `highlight` is the closest thing they find, but it wouldn't it make sense to have both?

* `highlight`: for painting the background with varying vertical bounds (no fancy stuff hard to reason about with changing heights)

* breakable `box`: meaningful borders for stroke, can affect layout with `inset` (and also `width`/`height`/`baseline`), simple (piecewise) rectangle shape making fancy stuff easy to reason about
